proxy/http/HttpConfig.cc

```diff
@@ -1065,14 +1140,54 @@ HttpConfig::startup()
+  RecRegisterConfigUpdateCb("proxy.config.http.connect_attempts_timeout_ms", &http_connect_attempt_timeout_cb, &c);
+  // Do initial load - use ms if set, otherwise use base if set, otherwise just leave it default.
+  if (!http_config_var_is_default("proxy.config.http.connect_attempts_timeout_ms")) {
+    if (REC_ERR_OKAY == RecGetRecordInt("proxy.config.http.connect_attempts_timeout_ms", &n)) {
+      c.oride.connect_attempts_timeout = HRTIME_MSECONDS(n);
+    }
+  } else if (!http_config_var_is_default("proxy.config.http.connect_attempts_timeout")) {
+    if (REC_ERR_OKAY == RecGetRecordInt("proxy.config.http.connect_attempts_timeout", &n)) {
+      c.oride.connect_attempts_timeout = HRTIME_SECONDS(n);
+    }
+  }
+
+  RecRegisterConfigUpdateCb("proxy.config.http.post_connect_attempts_timeout", &http_post_connect_attempt_timeout_cb, &c);
+  RecRegisterConfigUpdateCb("proxy.config.http.post_connect_attempts_timeout_ms", &http_post_connect_attempt_timeout_cb, &c);
+  // Do initial load - use ms if set, otherwise use base if set, otherwise just leave it default.
+  if (!http_config_var_is_default("proxy.config.http.post_connect_attempts_timeout_ms")) {
+    if (REC_ERR_OKAY == RecGetRecordInt("proxy.config.http.post_connect_attempts_timeout_ms", &n)) {
+      c.oride.post_connect_attempts_timeout = HRTIME_MSECONDS(n);
+    }
+  } else if (!http_config_var_is_default("proxy.config.http.post_connect_attempts_timeout")) {
+    if (REC_ERR_OKAY == RecGetRecordInt("proxy.config.http.post_connect_attempts_timeout", &n)) {
+      c.oride.post_connect_attempts_timeout = HRTIME_SECONDS(n);
+    }
+  }
+
   HttpEstablishStaticConfigLongLong(c.oride.parent_connect_attempts, "proxy.config.http.parent_proxy.total_connect_attempts");
   HttpEstablishStaticConfigLongLong(c.oride.parent_retry_time, "proxy.config.http.parent_proxy.retry_time");
   HttpEstablishStaticConfigLongLong(c.oride.parent_fail_threshold, "proxy.config.http.parent_proxy.fail_threshold");
   HttpEstablishStaticConfigLongLong(c.oride.per_parent_connect_attempts,
                                     "proxy.config.http.parent_proxy.per_parent_connect_attempts");
-  HttpEstablishStaticConfigLongLong(c.oride.parent_connect_timeout, "proxy.config.http.parent_proxy.connect_attempts_timeout");
+
+  RecRegisterConfigUpdateCb("proxy.config.http.parent_proxy.connect_attempts_timeout",
+                            &http_parent_proxy_connect_attempt_timeout_cb, &c);
+  RecRegisterConfigUpdateCb("proxy.config.http.parent_proxy.connect_attempts_timeout_ms",
+                            &http_parent_proxy_connect_attempt_timeout_cb, &c);
+  // Do initial load - use ms if set, otherwise use base if set, otherwise just leave it default.
+  if (!http_config_var_is_default("proxy.config.http.parent_proxy.connect_attempts_timeout_ms")) {
+    if (REC_ERR_OKAY == RecGetRecordInt("proxy.config.http.parent_proxy.connect_attempts_timeout_ms", &n)) {
+      c.oride.parent_connect_timeout = HRTIME_MSECONDS(n);
```

Comment:
I wonder if this should call `MillisecondsConverter(&c.oride.parent_connect_timeout, n)` for consistency.